api/osrd_infra/views/train_schedule/standalone_simulation_report.py

```diff
@@ -174,36 +173,56 @@ def build_signal_updates(signal_updates, departure_time):
     return results
 
 
-def project_signal_updates(signal_updates, projection_path_payload: PathPayload, departure_time):
+def project_signal_updates(route_occupancies, projection_path_payload: PathPayload, departure_time):
     results = []
 
-    updates_by_route_id = defaultdict(lambda: list())
-    for update in signal_updates:
-        for route_id in update["route_ids"]:
-            updates_by_route_id[route_id].append(update)
+    occupation_by_route_id = {}
+    for route_id, occupancy in route_occupancies.items():
+        occupation_by_route_id[route_id] = occupancy
 
     start_pos = 0
+    last_route = None
+    last_start_pos = 0
     for route_path in projection_path_payload.route_paths:
```

Comment:
If we do this by route, we will display one "canton" per route. That will make them very large with the new model, right?

Reply:
With the way I generate the routes. They can't be bigger than a track section.